libraries/chain/db_market.cpp

```diff
@@ -1210,10 +1278,14 @@ bool database::check_call_orders( const asset_object& mia, bool enable_black_swa
 
           if( before_core_hardfork_342 )
           {
-             order_receives = usd_to_buy * match_price; // round down, in favor of call order
+             limit_receives = usd_to_buy * match_price; // round down, in favor of call order
+             call_pays = limit_receives;
+          } else {
+             limit_receives = usd_to_buy.multiply_and_round_up( match_price ); // round up, in favor of limit order
+             call_pays      = usd_to_buy.multiply_and_round_up( call_pays_price ); // BSIP74; excess is fee.
+                                              // Note: TODO: Due to different rounding, couldn't this potentialy be
+                                              // one satoshi more than the blackswan check above? Can this bite us?
```

Comment:
@abitmore — Wanting to call attention to this question.  It's not a new issue — old code uses multiply_and_round_up seemingly without problem — but since we do a hard fork check above at line 1226 using multiply-and-round-down, is there a possibility that with the exact right price feed we could end up with call_pays being one satoshi more than the collateral in the call order?   I expect this would eventually cause an exception to be thrown, perhaps preventing a feed_update_operation or other operation that can move the call order price from confirming?